### ISSUE TITLE
VEGA VII system shutdowns within 30 seconds of running xmr stack

### ISSUE THREAD

**Issue by m03e5** [fireice-uk/xmr-stak#2244]
Brand new RADEON VEGA VII (Powercolor) Latest drivers available as of today from AMD website. WINDOWS 10 X64 HOME all updates done. Miner detects GPU properly and starts mining with speed approximately 1850hs on XMRV8 then system goes instant reboot within 30 seconds of running miner. System itself is running fine, mainboard, ram, psu are all good. I have ran two VEGA64 off this PC without any problems. CLEAN INSTALL OS. No vega registry edits, no overclocking, all stock, default settings. I have not yet tried block chain driver and not sure if it even work with this card, but will try if it will work then perhaps i will try registry edits, similar to 1 Gen VEGA.

# Basic information
  - Type of the CPU = VEGA RADEON VII (POWERCOLOR)
  - Type of the GPU (if you try to miner with the GPU).  66AF REV.C1  HBM2 16GB (MICRON)

# Compile issues
  - Which OS do you use? WINDOWS X64 HOME

# Issue with the execution
  - Do you compiled the miner by our own? = NO
  ```
# Stability issue
  - Is the CPU or GPU overclocked? = NO
  - Is the Main memory of the CPU or GPU undervolted? = NO
Any ideas?

**Comment by Spudz76** [comment]
No reason for that effect unless PSU is not enough, or worn out.  Risers or direct mobo?  Any adaptations in wiring, or real feeds direct from high quality PSU?  Watch out for melted connections at the PSU if its modular (usually the PSU end is not actually capable of 24/7 150w per whip without heating up).  Better to use non-modular supplies at least for the PCI-e whips.

Label spec was for day1, could be 66% of that depending on quality and usage and heat.  I have had several PSUs that work fine until one day when they decide the same old watts are too tough.  Had one that I used for months but I couldn't fire up any games, would reboot within 5 minutes as the PSU got hot and even worse at supplying watts required (well under label, probably 50%).

Even had some cheap ones that could barely cope with label draw on day1, heat overload tripped protection within the same half hour type timeframe.

**Comment by Spudz76** [comment]
Oh, and on the hashrate 1850 is about right for cn2v2 ("cryptonight_v8"/"monero" currencies) as the PoW is more difficult.

Those other higher numbers were from before the fork most likely (watch out for dates on posts!) as the old cn2v1 ("cryptonight_v7") was that speed on Vega (slightly less on the 56).

I think someone got a bit more out of one by undervolting and all that (keep it from hitting thermal or watt limiter).

**Comment by xq0404** [comment]
I have just ordered Radeon VII in the hopes of achieving up to 2800 H/s.  It seems to be more like a dream. Have you installed AMD's latest driver (win10-64bit-radeon-software-adrenalin-2019-edition-19.2.2-feb12)?

**Comment by m03e5** [comment]
@xq0404 
- this is the exact driver i have installed (win10-64bit-radeon-software-adrenalin-2019-edition-19.2.2-feb12)
@Spudz76 i know it s ounds like its PSU or what you have mentioned. Thats the first thought i had, i got both VEGA64's and original drive back to this pc and they run fine, so its not PSU. Cards go directly to board PCIE slots, no raisers or any adapters. PSU is not modular either. In terms of of hashing power 1850hs - its exact same as i get per VEGA64 on stack or cast miners. This is something i did not expect due to the fact that VEGA VII gpu core runs about 200+Mhz higher frequency AND VEGA VII has 16GB of HBM2 memory which makes me think it should get at least 2000hs on moneroV8. I think i have found problem, need to do a few tests to confirm, as of right now it looks like it may not be related to miner... will update. By the way here is videos how it does restart:
xmr stack: https://www.youtube.com/watch?v=FrwUAdloXZg
xmrig: https://www.youtube.com/watch?v=aDkurzmju8o

**Comment by psychocrypt** [comment]
it can still be the psu because vii is pulling over 300watt in the worst
case. Which powerplug do you use?

**Comment by psychocrypt** [comment]
I mean wich psu do you use?

**Comment by m03e5** [comment]
@psychocrypt 
- they PSU i have is Corsair TX850 and tried Cooler Master 1000W Silent Pro Gold - same behavior.

**Comment by psychocrypt** [comment]
mhh the point is that a program should be not able to crash the system if
it is running as user.if so than something else is broken.

**Comment by m03e5** [comment]
I have ran 2x VEGA64 off this Corsair TX850 without any problems. Same system. So its certainly not PSU or any other hardware. I am running FURMARK burn in test on VEGA VII in this system, been on for some time now and now reset. VEGA Fans are - 100% temp is ~76'c and junction is about 113'c, runs ok so far....

**Comment by wanginator** [comment]
@xq0404   Should be doable.  I got 2600h/s on mine with the launch day driver set with no tweaking.

**Comment by m03e5** [comment]
@wanginator
 - What model you have? Mine is POWERCOLOR AXVII16BHBM2-3DH

**Comment by wanginator** [comment]
@m03e5  I just used my vega config.  2 threads at 1932 each.  As for specs, all these cards are exactly identical.

**Comment by m03e5** [comment]
@wanginator 
- You mean vega56/64 ? with all registry edits etc ?

**Comment by wanginator** [comment]
@m03e5 yes, same stak config as 56/64.  No tweaks, powerplay tables, etc.  I literally copied my stak folder off of a vega rig, removed all but 2 threads and started stak just to see what it would do.

**Comment by xq0404** [comment]
I have just ordered Radeon VII in the hopes of achieving up to 2800 H/s.  It seems to be more like a dream. Have you installed AMD's latest driver (win10-64bit-radeon-software-adrenalin-2019-edition-19.2.2-feb12)?

> 
> 
> @xq0404 Should be doable. I got 2600h/s on mine with the launch day driver set with no tweaking.

Thank you! Good news!

**Comment by xq0404** [comment]
> 
> 
> @xq0404
> 
>     * this is the exact driver i have installed (win10-64bit-radeon-software-adrenalin-2019-edition-19.2.2-feb12)
>       @Spudz76 i know it s ounds like its PSU or what you have mentioned. Thats the first thought i had, i got both VEGA64's and original drive back to this pc and they run fine, so its not PSU. Cards go directly to board PCIE slots, no raisers or any adapters. PSU is not modular either. In terms of of hashing power 1850hs - its exact same as i get per VEGA64 on stack or cast miners. This is something i did not expect due to the fact that VEGA VII gpu core runs about 200+Mhz higher frequency AND VEGA VII has 16GB of HBM2 memory which makes me think it should get at least 2000hs on moneroV8. I think i have found problem, need to do a few tests to confirm, as of right now it looks like it may not be related to miner... will update. By the way here is videos how it does restart:
>       xmr stack: https://www.youtube.com/watch?v=FrwUAdloXZg
>       xmrig: https://www.youtube.com/watch?v=aDkurzmju8o
Suggested steps:
1. It seems you have another motherboard GPU Interl HD Graphics 510. I recommend disabling it;
2. deleting amd.tex, config.txt, cpu.txt, and rerunning the xmr-stak.exe to let it automatically reconfig;
3. reducing fan speed and noise level by sliding Power Limit to -40% and Min Acoustic Limit to 910. 
4. manually setting Max Temperature to 74-79 degrees Celsius. 
![rx480 driver](https://user-images.githubusercontent.com/20359673/52892460-920ea200-31cd-11e9-8242-6915e7458c01.JPG)

**Comment by MohitYadavGitHub** [comment]
> @xq0404 Should be doable. I got 2600h/s on mine with the launch day driver set with no tweaking.

Which memory manufacturer do have for the hbm? (Hynix , Micron, Samsung?)

I am getting 2400 with mine . Not sure if its the config or the card itself.

**Comment by raretechsea** [comment]
I wonder if the gpu heat-pad is not torqued properly?

**Comment by xq0404** [comment]
> Which memory manufacturer do have for the hbm? (Hynix , Micron, Samsung?)
> 
> I am getting 2400 with mine . Not sure if its the config or the card itself.
> 
> ALso a lot of rejected shares !

That's the minimum theoretically for XMR mining if VII gets 90MH/s for Ethereum.

**Comment by m03e5** [comment]
@xq0404 
 1 - i dont think it has any effect on vega performance... do you mean disabling it in device manager? Its cpu built in gfx, i dont think there is a way to disable it, except device manager.
 2 - did this, no change, actually hasrate dropped by approx 75-90hs, so reverted back to 2.8.2
 3 - noise/fan speed has nothing to do with performance AFAIK, it does not bother me.
 4 - a bit of noise but more stable performance is more important. see #3
 5 - you don't seem to have GPU Workload option under Global Graphics setting. It needs to be set at "Compute". - I have no idea where that setting is, checked everywhere it is not there... i have checked VEGA64 systems with blockchain drivers - that option is not there either. Can you do screenshot of it and what driver version/card brand do you use? I have a feeling this has something to do with performance.
P.S. Vega VII does 90HS on ETH, i can confirm this.

**Comment by xq0404** [comment]
> 
> 
> @xq0404
> 1 - i dont think it has any effect on vega performance... do you mean disabling it in device manager? Its cpu built in gfx, i dont think there is a way to disable it, except device manager.
> 2 - did this, no change, actually hasrate dropped by approx 75-90hs, so reverted back to 2.8.2
> 3 - noise/fan speed has nothing to do with performance AFAIK, it does not bother me.
> 4 - a bit of noise but more stable performance is more important. see #3
> 5 - you don't seem to have GPU Workload option under Global Graphics setting. It needs to be set at "Compute". - I have no idea where that setting is, checked everywhere it is not there... i have checked VEGA64 systems with blockchain drivers - that option is not there either. Can you do screenshot of it and what driver version/card brand do you use? I have a feeling this has something to do with performance.
> P.S. Vega VII does 90HS on ETH, i can confirm this. (this card does 87-88Hs)

Yes, only devince manger can disable Interl HD Graphics 510. "GPU Workload" option" is under Global Graphics setting, perhaps it's because mine is an RX 480 GPU.  At present, mining ethereum is much way more profitiable than mining Monero.  Actually while you mine Ethereum with Nanominer, you can simultaneously mine Monero with xmr-stak (by disabling GPU mining in the amd.txt and using CPU only).

**Comment by m03e5** [comment]
So after some tinkering with settings and overclock i was able to reach 2007hs on 2.8.3 stack..
![vega vii 2007hs](https://user-images.githubusercontent.com/34433539/53109409-0c089900-34f6-11e9-9954-7542f8c119c2.jpg)

**Comment by psychocrypt** [comment]
most user using single thread for cn-gpu. Try if you see maybe the same
hashrate with one thread

**Comment by m03e5** [comment]
I tried running miner in 1 thread - same speed. Not sure what the problem is... I did manage to get 2700-2800hs on cast. The fastest rate i have seen is 2910hs which still less than other people report (3100-3200hs on cnv8)
![vega 7 2700](https://user-images.githubusercontent.com/34433539/53278613-7e7ca300-36c7-11e9-8f68-e3f0b572b66e.jpg)

**Comment by psychocrypt** [comment]
now I am confused. You posted a Screenshot from the xmrig miner not
xmr-stak.
Is this still related to the crash after 30 sec? Do not mix issues. If this
is now a performance releated issues please ooen a new issue.

**Comment by m03e5** [comment]
@psychocrypt  
- my apologies. The shutdown problem, as appears, related to defective video card, not the miner. This question is figured out now.

**Comment by psychocrypt** [comment]
thx for the info. hope you will get a new card soon

**Comment by fractalyse** [comment]
I'm experiencing similar issue on ubuntu 18.04.
I'm running the vega @ 1135Mhz with a power draw of 80w, and my computer randomly reboot, card is running at 60°C

**Comment by psychocrypt** [comment]
The miner is not able to reboot your OS. Even if there would be an issue
the miner should never be able to crash your system since it run in user
mode.
It sounds like issue with the power supply.
Do you connected all power plugs to the card?

**Comment by fractalyse** [comment]
All cable are connected, I'm using EVGA 1000w T2.
It's not the PSU I think beacause I can benchmark for hours without issue with a power draw of 250w.. never shutdown.

This is related to compute job. However, I can stress the card for hours using LuxMark it will simply never stop. Same in openCL benchmark..

This only happens with miners, xmr-stak, xmr-rig, phoenixminer and so on..
